opnode/rollup/driver/step.go

```diff
@@ -283,6 +291,32 @@ func (d *outputImpl) insertHeadBlock(ctx context.Context, fc l2.ForkchoiceState,
 	if err != nil {
 		return nil, fmt.Errorf("failed to get execution payload: %w", err)
 	}
+	// Sanity check payload before inserting it
+	if len(payload.TransactionsField) == 0 {
+		return nil, errors.New("no transactions in returned payload")
+	}
+	txns := payload.Transactions()
+	if txns[0].Type() != types.DepositTxType {
+		return nil, fmt.Errorf("first transaction was not deposit tx. Got %v", txns[0].Type())
+	}
+	// Ensure that the deposits are first
+	var lastDeposit int
+	doneWithDeposits := false
```

Comment:
instead of adding the extra var it may be nice to just determine `lastDeposit` first (and maybe deduplicate that code with the above `createNewBlock` func) and then run a 2nd loop like `for i, tx := range txns[lastDeposit+1:] { ` to check that all remaining txs are not a deposit.